drivers/hmis/app/models/hmis/hud/processors/base.rb

```diff
@@ -0,0 +1,22 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+class Hmis::Hud::Processors::Base
+  def initialize(processor)
+    @processor = processor
+  end
+
+  def hud_name(field)
+    field.underscore
+  end
+
+  def hud_type(field)
+    type = schema.fields[field].type
```

Comment:
where is `schema` coming from?

Reply:
it's the GraphQL type, defined on each of the subclasses of `Hmis::Hud::Processors::Base`

Reply:
Thanks.  Should this be defined as an abstract class then?

Reply:
Probably -- but these are utility classes, `self.abstract_class = true` is a AR construct. I'll put in
```
def schema
  raise 'Implement in sub-class'
end
```